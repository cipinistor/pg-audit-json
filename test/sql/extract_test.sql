CREATE EXTENSION "pg-audit-json";

-- Extract json for one key
SELECT jsonb_extract('{"a": 1, "b": 2, "c": 3}'::jsonb, '{a}'::text[]);

-- Extract json for multiple keys
SELECT jsonb_extract('{"a": 1, "b": 2, "c": 3}'::jsonb, '{a,c}'::text[]);

-- Extract json for no key 
SELECT jsonb_extract('{"a": 1, "b": 2, "c": 3}'::jsonb, '{}'::text[]);

DROP EXTENSION "pg-audit-json";
