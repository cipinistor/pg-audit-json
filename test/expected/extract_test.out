CREATE EXTENSION "pg-audit-json";
-- Extract json for one key
SELECT jsonb_extract('{"a": 1, "b": 2, "c": 3}'::jsonb, '{a}'::text[]);
 jsonb_extract 
---------------
 {"a": 1}
(1 row)

-- Extract json for multiple keys
SELECT jsonb_extract('{"a": 1, "b": 2, "c": 3}'::jsonb, '{a,c}'::text[]);
  jsonb_extract   
------------------
 {"a": 1, "c": 3}
(1 row)

-- Extract json for no key 
SELECT jsonb_extract('{"a": 1, "b": 2, "c": 3}'::jsonb, '{}'::text[]);
 jsonb_extract 
---------------
 
(1 row)

DROP EXTENSION "pg-audit-json";
